### PR TITLE
fix GitHub errors; save `overmap_fast_travel` & `overmap_fast_scroll`

### PR DESCRIPTION
**Pull request by Brambor**
#### Summary
None

#### Purpose of change

`overmap_fast_travel` & `overmap_fast_scroll` should save now. ~(TODO: check)~

Fix some Github Errors:

<details>

<summary>
Images of errors
</summary>

image

image

image

</details>

The errors/warnings can be seen here (for example):
 - #76141

#### Describe the solution

Also removed the comment. It didn't seem helpful. Otherwise, the line would have to split for Astyle.

#### Testing

1. Compiled locally.
2. ~Let GitHub finish tests.~ GitHub doesn't report the test failures in this PR.
3. Check `overmap_fast_travel` & `overmap_fast_scroll` should save now.
     1. Load a save.
     2. Open `m`ap.
     3. See that fast scroll is off.
     4. Toggle it.
     5. Save game.
     6. Restart CDDA.
     7. Load a save.
     8. Open `m`ap.
     9. See that fast scroll is on (in new version) off (in old version).

#### Additional context

### CODE REVIEW

File: src/npctalk_funcs.cpp [hunk 1]
```diff
@@ -1285,8 +1285,8 @@ void talk_function::distribute_food_auto( npc &p )
     zone_manager &mgr = zone_manager::get_manager();
     const tripoint_abs_ms &npc_abs_loc = p.get_location();
     // 3x3 square with NPC in the center, includes NPC's tile and all adjacent ones, for overflow
-    const tripoint top_left = npc_abs_loc.raw() + point{-1, -1}; // Awful hack, zones want the raw value
```
Comment (RenechCDDA) [inline src/npctalk_funcs.cpp:1288]: It is still awful and notable that zone_manager::add() is expecting a raw tripoint that is actually a tripoint_abs_ms

Comment (Brambor) [reply src/npctalk_funcs.cpp:1288]: Added `// TODO: fix point types; Awful hack, zones want the raw value`

**Comment by Brambor** [comment]
I tested that the values save. Good to go!

**Comment by Brambor** [comment]
Oh, the failing test is required, restarting then...

**Pull request merged**